### PR TITLE
Added service that use Razor. Added the email templates path structure

### PR DESCRIPTION
**Pull request by RBiyovskiy**

### CODE REVIEW

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Services/EmailTemplates.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿namespace OutOfSchool.RazorTemplatesData.Services
+{
+    public static class EmailTemplates
+    {
+        /// <summary>
+        /// Bass path to the html email templates
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Services/EmailTemplates.cs:6]: I hope in `/// Bass path` you mean `Base path`. Please correct here and in `MainPathPlainText`.

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Services/EmailTemplates.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        /// </summary>
+        private static string MainPath => "/Views/Emails/";
+
+        /// <summary>
+        /// Bass path to the plain text email templates
+        /// </summary>
+        private static string MainPathPlainText => "/ViewsPlainText/Emails/";
+
+        /// <summary>
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Services/EmailTemplates.cs:15]: This comment block is related only to first property, while is describing 4 properties, please rewrite it in simple `// ` comment, or add personal comments (though I see no necessity in commenting them at all)

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/ChangeEmail/ChangeEmail.cshtml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+<br />
+
+<p>
+    If you have any questions, just reply to this email—we're always happy to help out.
+</p>
+
+<br />
+
+<p>
+    The Contoso Team
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/ChangeEmail/ChangeEmail.cshtml:30]: `The Contoso Team` - looks like legacy code that should be changed

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+﻿@*
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml:1]: Really strange comment

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Shared/EmailLayout.cshtml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <title></title>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <style type="text/css">
+        /* FONTS */
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Shared/EmailLayout.cshtml:9]: Another comments that should be swept out

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Services/RazorViewToStringRenderer.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OutOfSchool.RazorTemplatesData.Services
+{
+    public class RazorViewToStringRenderer : IRazorViewToStringRenderer
+    {
+        private readonly IRazorViewEngine _viewEngine;
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Services/RazorViewToStringRenderer.cs:19]: Private field naming seems to be inconsistent to other code in project. Please confirm if underscore (_) should be used as prefix.

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml [hunk 7]
```diff
@@ -1,5 +1,25 @@
...
+
+<p>Пароль:  @Model.Password</p>
+
+<br />
+
+@await Html.PartialAsync("EmailButton", new EmailButtonViewModel("Confirm Email", Model.ConfirmationUrl))
+
+<br />
+
+<p>If you have any questions, just reply to this email—we're always happy to help out.</p>
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml:25]: Please translate last paragraph to Ukrainian or remove it

File: OutOfSchool/OutOfSchool.RazorTemplatesData/ViewsPlainText/Emails/NewAdminInvitation/NewAdminInvitation.cshtml [hunk 8]
```diff
@@ -1,5 +1,16 @@
...
+
+Для реєстрації на платформі перейдіть за посиланням та заповність ваші данні в особистому кабінеті.
+
+@Model.ConfirmationUrl
+
+Логін: @Model.Email
+
+Пароль:  @Model.Password
+
+If you have any questions, just reply to this email—we're always happy to help out.
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/ViewsPlainText/Emails/NewAdminInvitation/NewAdminInvitation.cshtml:16]: Please don't mix two languages, either translate last paragraph or drop it

File: OutOfSchool/OutOfSchool.RazorTemplatesData/ViewsPlainText/Emails/NewAdminInvitation/NewAdminInvitation.cshtml [hunk 9]
```diff
@@ -1,5 +1,16 @@
...
-    For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
-*@
+﻿@using OutOfSchool.RazorTemplatesData.Models.Emails
+@model AdminInvitationViewModel
+
 @{
+    ViewData["EmailTitle"] = "Welcome!";
 }
+
+Для реєстрації на платформі перейдіть за посиланням та заповність ваші данні в особистому кабінеті.
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/ViewsPlainText/Emails/NewAdminInvitation/NewAdminInvitation.cshtml:8]: Could you fix the mistype `заповність` => `заповніть`

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml [hunk 10]
```diff
@@ -1,5 +1,25 @@
...
-*@
+﻿@using OutOfSchool.RazorTemplatesData.Models.Emails
+@using OutOfSchool.RazorTemplatesData.Models.Shared
+@model AdminInvitationViewModel
+
 @{
+    ViewData["EmailTitle"] = "Welcome!";
 }
+
+<p>Для реєстрації на платформі перейдіть за посиланням та заповність ваші данні в особистому кабінеті.</p>
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml:9]: Please fix spelling: `заповність` => `заповніть`, `данні` => `дані`

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml (hunk 7 above)
Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.RazorTemplatesData/Views/Emails/NewAdminInvitation/NewAdminInvitation.cshtml:25]: Ok, but please don't increase entropy by adding mistakes from the very beginning. 
I'm bit afraid of situation that mentioned frontend team might blindly reuse you texts and thus error will slip into final version of code.

File: OutOfSchool/OutOfSchool.RazorTemplatesData/Services/RazorTemplates.cs [hunk 11]
```diff
@@ -0,0 +1,31 @@
+﻿namespace OutOfSchool.RazorTemplatesData.Services
+{
+    public static class RazorTemplates
+    {
+        /// <summary>
+        /// Base path to the html email templates
+        /// </summary>
+        private static string MainPath => "/Views/Emails/";
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.RazorTemplatesData/Services/RazorTemplates.cs:8]: Consider if `MainPath` should be renamed to something like `MainEmailPath`, Same for `MainPathPlainText`

Comment (RBiyovskiy) [reply OutOfSchool/OutOfSchool.RazorTemplatesData/Services/RazorTemplates.cs]: Renamed it, but I'm not sure it's critical. I think it is clear from the context what the main path is.

**Review by VadymLevkovskyi — APPROVED** [review]

**Pull request merged**